### PR TITLE
Fix for Logistic Regression loss scaling

### PR DESCRIPTION
**Pull request by Alexsandruss**
### Description 

Solves https://github.com/autogluon/autogluon/issues/4280.

LogReg loss scaling was changed from 1.3 to 1.4 sklearn version in https://github.com/scikit-learn/scikit-learn/pull/26721 resulting in worse convergence of sklearnex LogReg compared to stock sklearn and appearing in mentioned AutoGluon issue.
Sklearnex LogReg coefficient became different due to change of convergence way, so few deselections of tests are needed because of that.

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by avolkov-intel** [comment]
I believe we should run Logistic Regression benchmarks to ensure this update does not cause performance degradation

**Comment by Alexsandruss** [comment]
image

No significant change except number of iterations and logloss which was different compared to stock sklearn before.

**Review by avolkov-intel — APPROVED** [review]

File: deselected_tests.yaml [hunk 1]
```diff
@@ -190,6 +190,13 @@ deselected_tests:
   # margin above the test threshold, see https://github.com/scikit-learn/scikit-learn/pull/13645
   - linear_model/tests/test_logistic.py::test_dtype_match
 
+  # Logistic Regression coeffs change due to fix for loss scaling
+  # (https://github.com/scikit-learn/scikit-learn/pull/26721)
+  - feature_selection/tests/test_from_model.py::test_importance_getter[estimator0-named_steps.logisticregression.coef_]
+  - inspection/_plot/tests/test_boundary_decision_display.py::test_class_of_interest_binary[predict_proba]
+  - linear_model/tests/test_sag.py::test_sag_pobj_matches_logistic_regression[csr_matrix]
+  - linear_model/tests/test_sag.py::test_sag_pobj_matches_logistic_regression[csr_array]
```
Comment (samir-nasibli) [inline deselected_tests.yaml:198]: Please make sure that we are covering these deselections with our baclkog

Comment (Alexsandruss) [reply deselected_tests.yaml:198]: Done.

**Review by samir-nasibli — APPROVED** [review]
@Alexsandruss thank you for the updates and sharing numbers.
Looks good to me. Assuming green CI checks

**Comment by Alexsandruss** [comment]
Windows failures are unrelated to this PR, see current public CI: https://github.com/intel/scikit-learn-intelex/runs/27052083355

**Comment by napetrov** [comment]
@mergify backport rls/2024.6.0-rls
